staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/validation_test.go

```diff
@@ -182,9 +182,38 @@ func TestValidateNestedValueValidationComplete(t *testing.T) {
 		x := reflect.ValueOf(&vv.ForbiddenExtensions).Elem()
 		fuzzer.Fuzz(x.Field(i).Addr().Interface())
 
-		errs := validateNestedValueValidation(vv, false, false, fieldLevel, nil)
+		errs := validateNestedValueValidation(vv, false, false, fieldLevel, nil, ValidationOptions{})
 		if len(errs) == 0 && !reflect.DeepEqual(vv.ForbiddenExtensions, Extensions{}) {
 			t.Errorf("expected ForbiddenExtensions validation errors for: %#v", vv)
 		}
 	}
+
+	for allowed := false; allowed != true; allowed = true {
+		opts := ValidationOptions{
+			AllowNestedXValidations:                           allowed,
+			AllowNestedAdditionalProperties:                   allowed,
+			AllowValidationPropertiesWithAdditionalProperties: allowed,
+		}
```

Comment:
I'm not sure a matrix adds much value since the the flags are meant to be orthogonal toggles for a single class of error, but I've changed it anyway since there were only two cases.

I also added a separate test for completeness which did not exist earlier, to test the `AllowValidationPropertiesWithAdditionalProperties` flag since it is thrown by a different internal function